src/ponyfills/arrays.js

```diff
@@ -0,0 +1,216 @@
+    //      with argument 0.
+    // c. ReturnIfAbrupt(A).
+    A = isCallable(C) ? Object(new C()) : [];
+
+    // d. Let iterator be GetIterator(items, usingIterator).
+    var iterator = usingIterator.call(items);
+
+    // e. ReturnIfAbrupt(iterator).
+    if (iterator == null) {
+      throw new TypeError(
+        'Array.from requires an array-like or iterable object'
+      );
+    }
+
+    // f. Let k be 0.
+    k = 0;
+
+    // g. Repeat
+    var next, nextValue;
+    while (true) { // eslint-disable-line
+      // i. Let Pk be ToString(k).
+      // ii. Let next be IteratorStep(iterator).
+      // iii. ReturnIfAbrupt(next).
+      next = iteratorStep(iterator);
+
+      // iv. If next is false, then
+      if (!next) {
+
+        // 1. Let setStatus be Set(A, "length", k, true).
+        // 2. ReturnIfAbrupt(setStatus).
+        A.length = k;
+
+        // 3. Return A.
+        return A;
+      }
+      // v. Let nextValue be IteratorValue(next).
+      // vi. ReturnIfAbrupt(nextValue)
+      nextValue = next.value;
+
+      // vii. If mapping is true, then
+      //   1. Let mappedValue be Call(mapfn, T, «nextValue, k»).
+      //   2. If mappedValue is an abrupt completion, return
+      //      IteratorClose(iterator, mappedValue).
+      //   3. Let mappedValue be mappedValue.[[value]].
+      // viii. Else, let mappedValue be nextValue.
+      // ix.  Let defineStatus be the result of
+      //      CreateDataPropertyOrThrow(A, Pk, mappedValue).
+      // x. [TODO] If defineStatus is an abrupt completion, return
+      //    IteratorClose(iterator, defineStatus).
+      if (mapFn) {
+        A[k] = mapFn.call(T, nextValue, k);
+      }
+      else {
+        A[k] = nextValue;
+      }
+      // xi. Increase k by 1.
+      k++;
+    }
+    // 7. Assert: items is not an Iterable so assume it is
+    //    an array-like object.
+  } else {
+
+    // 8. Let arrayLike be ToObject(items).
+    var arrayLike = Object(items);
+
+    // 9. ReturnIfAbrupt(items).
+    if (items == null) {
+      throw new TypeError(
+        'Array.from requires an array-like object - not null or undefined'
+      );
+    }
+
+    // 10. Let len be ToLength(Get(arrayLike, "length")).
+    // 11. ReturnIfAbrupt(len).
+    var len = toLength(arrayLike.length);
+
+    // 12. If IsConstructor(C) is true, then
+    //     a. Let A be Construct(C, «len»).
+    // 13. Else
+    //     a. Let A be ArrayCreate(len).
+    // 14. ReturnIfAbrupt(A).
+    A = isCallable(C) ? Object(new C(len)) : new Array(len);
+
+    // 15. Let k be 0.
+    k = 0;
+    // 16. Repeat, while k < len… (also steps a - h)
+    var kValue;
+    while (k < len) {
+      kValue = arrayLike[k];
+      if (mapFn) {
+        A[k] = mapFn.call(T, kValue, k);
+      }
+      else {
+        A[k] = kValue;
+      }
+      k++;
+    }
+    // 17. Let setStatus be Set(A, "length", len, true).
+    // 18. ReturnIfAbrupt(setStatus).
+    A.length = len;
+    // 19. Return A.
+  }
+  return A;
+}
```

Comment:
This seems like a lot of weight for, as far as I can, just `Array.from()`. I have a smaller shim in execution plan here:
https://github.com/SentientTechnologies/execution-plan/blob/master/src/shared/shims/from-array.shim.ts

Reply:
Mine doesn't account of iterables yet though if you need those.

Reply:
I use it to turn a set to an array so i think i need it to account for iterables

Reply:
I reduced it and used your implementation from the link you supplied. I didnt get what you meant by iterables as first

Reply:
Done.